test/test_namedtensor.py

```diff
@@ -26,6 +26,14 @@ class TestNamedTensor(TestCase):
     def test_trivial(self):
         pass
 
+    # TODO(rzou): Some form of this check should be added to self.assertEqual.
+    # Right now I don't know what it should look like.
+    def assertTensorDataAndNamesEqual(self, x, y):
+        self.assertEqual(x.names, y.names)
+        unnamed_x = x.set_names(None)
+        unnamed_y = y.set_names(None)
+        self.assertEqual(unnamed_x, unnamed_y)
```

Comment:
if `assertEqual` doesn't check names, why do we need to make these unnamed?
if `assertEqual` does compare names, why do we need this new method?

Should this check for named tensors being enabled in the config before calling set_names?

I think I didn't realize `set_names` was returning a new tensor instead of mutating the passed tensor, which was why I was suggesting `names_`. Still not sure what we should be calling it though, if it's not a mutator then `set` doesn't seem right either.

Reply:
> if assertEqual doesn't check names, why do we need to make these unnamed?
if assertEqual does compare names, why do we need this new method?

`assertEqual` doesn't support named tensors because it calls some operations that don't have named inference rules implemented. In the long run it will support named tensors and it should probably check names.

The reason why I am unsure is because it does some strange behavior in other cases. One example is `assertEqual(torch.ones(3, dtype=torch.float), torch.ones(3, dtype=torch.int))` is True.

> Should this check for named tensors being enabled in the config before calling set_names?

I don't think so. This check is exclusive to TestNamedTensor, which can't be run unless named tensors are enable.

> I think I didn't realize set_names was returning a new tensor instead of mutating the passed tensor, which was why I was suggesting names_. Still not sure what we should be calling it though, if it's not a mutator then set doesn't seem right either.

Yeah, this question has been on my mind for a while. `names_` is a good name for the mutator. However, we need a name for the out-of-place variant.